### PR TITLE
Test goal range in "only" selectors

### PR DESCRIPTION
**Pull request by SkySkimmer**
I don't know what goal_selector.v was supposed to test but CI says
nobody relied on it.

### CODE REVIEW

File: engine/proofview.ml [hunk 1]
```diff
@@ -373,32 +373,26 @@ let tclTRYFOCUS i j t = tclFOCUS ~nosuchgoal:(tclUNIT ()) i j t
...
-      else if i > n then None
-      else if j < 1 then None
-      else Some ((max i 1), (min j n))
-    in
-    let l = CList.map_filter sanitize l in
+  let n = CList.length comb in
+  (* First, remove empty intervals, and bound the intervals to the number
+     of goals. *)
+  let ok (i, j) = 1 <= i && i <= j && j <= n in
+  if not (CList.for_all ok l) then tclZERO (NoSuchGoals 0)
```
Comment (Zimmi48) [inline engine/proofview.ml]: The comment above becomes inaccurate since we don't remove empty intervals, we raise an error if there are some.
Furthermore, it seems a bit strange to decide to raise a NoSuchGoal error if the problem is a nonsensical interval.
Finally, why use `tclZERO (NoSuchGoals 0)` here instead of `nosuchgoal`?

**Review by Zimmi48 — COMMENTED** [review]
You didn't add any test case with empty intervals.

File: engine/proofview.ml (hunk 1 above)
Comment (ppedrot) [reply engine/proofview.ml]: I am fine with the `NoSuchGoal` error for nonsensical intervals, but indeed maybe using the generalized error is probably better.

Comment (SkySkimmer) [reply engine/proofview.ml]: NoSuchGoal is there only because this is rebased from when I first wrote it, which was before nosuchgoal

Comment (SkySkimmer) [reply engine/proofview.ml]: >Furthermore, it seems a bit strange to decide to raise a NoSuchGoal error if the problem is a nonsensical interval.

nosuchgoal could be called "invalid selector" instead I guess
I don't care enough to make a PR for it though

**Review by Zimmi48 — APPROVED** [review]

**Comment by ppedrot** [comment]
@SkySkimmer can you rebase to get a fresh CI?

**Review by ppedrot — APPROVED** [review]
